### PR TITLE
feat(Testcase) : Adding Daemonset & Statefulsets creation testcase(Co…

### PR DESCRIPTION
**Pull request by prashantvicky**
…mments Incorporated)
@ewiseblatt 
In order to re base the Pull request to master i have deleted the old one and merged my code in it.
This pull request is in continuation to the old pull request (https://github.com/spinnaker/spinnaker/pull/2061) 
Please close the old Pull request No: 2061 and consider this one.
As i am a java/Groovy developer so forgive me for the poor indentation as i have just started python.
Please let me know if there is any thing else
Reagrds
Prashant    

We prefer small, well tested pull requests.

Please refer to [Contributing to Spinnaker](https://spinnaker.io/community/contributing/).

When filling out a pull request, please consider the following:

* Follow the commit message conventions [found here](http://www.spinnaker.io/v1.0/docs/how-to-submit-a-patch).
* Provide a descriptive summary for your changes.
* If it fixes a bug or resolves a feature request, be sure to link to that issue.
* Add inline code comments to changes that might not be obvious.
* Squash your commits as you keep adding changes.
* Add a comment to @spinnaker/reviewers for review if your issue has been outstanding for more than 3 days.

Note that we are unlikely to accept pull requests that add features without prior discussion. The best way to propose a feature is to open an issue first and discuss your ideas there before implementing them.

### CODE REVIEW

File: testing/citest/tests/kube_smoke_test.py [hunk 1]
```diff
@@ -390,6 +390,190 @@ def make_deploy_stage(self, imageSource=None, requisiteStages=[], **kwargs):
...
 
+  def create_statefulset_pipeline(self):
+      bindings = self.bindings
+
+      payload = self.agent.make_json_payload_from_kwargs(
+          job=[{
+              'type': 'manual',
+              'user': '[anonymous]'
+          }],
+          description='Create StatefulSet'.format(
```
Comment (ewiseblatt) [inline testing/citest/tests/kube_smoke_test.py]: You have a format here but no parameters in the string to inject into.

File: testing/citest/tests/kube_smoke_test.py [hunk 2]
```diff
@@ -390,6 +390,190 @@ def make_deploy_stage(self, imageSource=None, requisiteStages=[], **kwargs):
...
+      expect_match = {key: jp.EQUIVALENT(value)
+                      for key, value in job.items()}
+      expect_match['stages'] = jp.LIST_MATCHES(
+          [
+              jp.DICT_MATCHES({key: jp.EQUIVALENT(value)
+                               for key, value in deploy_stage.items()})])
+
+      builder = st.HttpContractBuilder(self.agent)
+      (builder.new_clause_builder('Has Pipeline',
+                                  retryable_for_secs=15)
```
Comment (ewiseblatt) [inline testing/citest/tests/kube_smoke_test.py]: I think 15 is good here, though you have 60 above where I think they underlying asynchronous stability issues (I assume eventual consistency in front50) are the same. You have 15 below as well so perhaps the original should be 15.

File: testing/citest/tests/kube_smoke_test.py [hunk 3]
```diff
@@ -390,6 +390,190 @@ def make_deploy_stage(self, imageSource=None, requisiteStages=[], **kwargs):
...
+                                  contract=builder.build())
+
+  def create_daemonset_pipeline(self):
+      bindings = self.bindings
+      payload = self.agent.make_json_payload_from_kwargs(
+          job=[{
+              'type': 'manual',
+              'user': '[anonymous]'
+          }],
+          description='Create Daemonset'.format(
```
Comment (ewiseblatt) [inline testing/citest/tests/kube_smoke_test.py]: injecting parameter values into string that has no parameters.

File: testing/citest/tests/kube_smoke_test.py [hunk 4]
```diff
@@ -390,6 +390,190 @@ def make_deploy_stage(self, imageSource=None, requisiteStages=[], **kwargs):
...
+          }],
+          description='Create StatefulSet'.format(
+              self.__lb_name,
+              bindings['SPINNAKER_KUBERNETES_ACCOUNT']),
+          application=self.TEST_APP)
+      builder = kube.KubeContractBuilder(self.kube_observer)
+      (builder.new_clause_builder('StatefulSet Added', retryable_for_secs=60)
+       .get_resources(
+          'statefulsets', extra_args=[self.TEST_APP, '--namespace', self.TEST_NAMESPACE])
+       .contains_path_value('kind', "StatefulSet"))
```
Comment (ewiseblatt) [inline testing/citest/tests/kube_smoke_test.py]: Single quote the string.

File: testing/citest/tests/kube_smoke_test.py [hunk 5]
```diff
@@ -390,6 +390,190 @@ def make_deploy_stage(self, imageSource=None, requisiteStages=[], **kwargs):
...
+          }],
+          description='Create Daemonset'.format(
+              self.__lb_name,
+              bindings['SPINNAKER_KUBERNETES_ACCOUNT']),
+          application=self.TEST_APP)
+      builder = kube.KubeContractBuilder(self.kube_observer)
+      (builder.new_clause_builder('Daemonset Added', retryable_for_secs=60)
+       .get_resources(
+          'daemonsets', extra_args=[self.TEST_APP, '--namespace', self.TEST_NAMESPACE])
+       .contains_path_value('kind', "DaemonSet"))
```
Comment (ewiseblatt) [inline testing/citest/tests/kube_smoke_test.py]: Single quote string

**Comment by prashantvicky** [comment]
@ewiseblatt 
I have added all the changes you have asked , Please review.
Regards
Prashant

**Comment by prashantvicky** [comment]
@ewiseblatt 
Added Space as requested.

**Comment by prashantvicky** [comment]
@ewiseblatt can you please merge if all gud.
Regards
Prashant

**Comment by lwander** [comment]
@prashantvicky this does not test delete, and is not passing on our integration test server as a result. When the test completes, it's unable to delete the application these resources belong to.

**Pull request merged**